public/app/plugins/datasource/loki/datasource.ts

```diff
@@ -384,6 +386,37 @@ export class LokiDatasource
     return res.data || [];
   }
 
+  async queryStatsRequest(query: LokiQuery): Promise<QueryStats> {
+    const { start, end } = this.getTimeRangeParams();
+
+    const labelMatcherPositions = getStreamSelectorPositions(query.expr);
+    const labelMatchers = labelMatcherPositions.map((labelMatcher) => {
+      return query.expr.slice(labelMatcher.from, labelMatcher.to);
+    });
+
+    let statsForAll: QueryStats = { streams: 0, chunks: 0, bytes: 0, entries: 0 };
+
+    for (const labelMatcher of labelMatchers) {
+      const { data } = await lastValueFrom(
```

Comment:
Here instead of `getBackendSrv().fetch<QueryStats>` I'd suggest to use `this.metadataRequest` in the same way as we use it to fetch labels, values, test data source connection,... . The `this.metadataRequest` uses `this.getResource` method that calls correct backend endpoint which is `/api/datasources/24/resources/...` . In your case, it is using `api/datasources/proxy/24/...` endpoint, which endpoint we migrated from, because we want all resources to be run trough `/resources`.